packages/flutter_tools/lib/executable.dart

```diff
@@ -124,69 +133,85 @@ Future<Null> main(List<String> args) async {
+      return _exit(1);
+    } else {
+      if (error is String)
+        stderr.writeln('Oops; flutter has exited unexpectedly: "$error".');
+      else
+        stderr.writeln('Oops; flutter has exited unexpectedly.');
+
+      await CrashReportSender.instance.sendReport(error: error, stackTrace: chain);
+      try {
+        File file = await _createLocalCrashReport(args, error, chain);
+        stderr.writeln(
+            'Crash report written to ${file.path};\n'
+                'please let us know at https://github.com/flutter/flutter/issues.',
+        );
+        return _exit(1);
+      } catch (error) {
         stderr.writeln(
-            "Run 'flutter -h' (or 'flutter <command> -h') for available "
-                "flutter commands and options."
+            'Unable to generate crash report due to secondary error: $error\n'
+                'please let us know at https://github.com/flutter/flutter/issues.',
         );
-        // Argument error exit code.
-        _exit(64);
-      } else if (error is ToolExit) {
-        if (error.message != null)
-          stderr.writeln(error.message);
-        if (verbose) {
-          stderr.writeln();
-          stderr.writeln(chain.terse.toString());
-          stderr.writeln();
-        }
-        _exit(error.exitCode ?? 1);
-      } else if (error is ProcessExit) {
-        // We've caught an exit code.
-        _exit(error.exitCode);
-      } else {
-        // We've crashed; emit a log report.
-        stderr.writeln();
-
-        flutterUsage.sendException(error, chain);
-
-        if (isRunningOnBot) {
-          // Print the stack trace on the bots - don't write a crash report.
-          stderr.writeln('$error');
-          stderr.writeln(chain.terse.toString());
-          _exit(1);
-        } else {
-          if (error is String)
-            stderr.writeln('Oops; flutter has exited unexpectedly: "$error".');
-          else
-            stderr.writeln('Oops; flutter has exited unexpectedly.');
-
-          _createCrashReport(args, error, chain).then<Null>((File file) {
-            stderr.writeln(
-              'Crash report written to ${file.path};\n'
-              'please let us know at https://github.com/flutter/flutter/issues.',
-            );
-            _exit(1);
-          }).catchError((dynamic error) {
-            stderr.writeln(
-              'Unable to generate crash report due to secondary error: $error\n'
-              'please let us know at https://github.com/flutter/flutter/issues.',
-            );
-            _exit(1);
-          });
-        }
+        return _exit(1);
       }
-    });
-  });
+    }
+  }
 }
 
-Future<File> _createCrashReport(List<String> args, dynamic error, Chain chain) async {
-  FileSystem fs = const LocalFileSystem();
+/// Saves the crash report to a local file.
+Future<File> _createLocalCrashReport(List<String> args, dynamic error, Chain chain) async {
+  FileSystem fs = context?.getVariable(FileSystem);
```

Comment:
Since you don't want to write local crashes in your test, maybe make `reportCrashes` an enum value with `{LOCAL, SERVER, BOTH, NONE}` values?